File: python/encode_and_decode_strings.py
class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """

        res = ''
        for s in strs:
            res += str(len(s)) + '#' + s
        print(res)
        return res

        

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """
        i = 0
        res = []

        while i < len(s):
            j = i
            while j < len(s) and s[j] != '#':
                j += 1
            lenWord = int(s[i:j])
            res.append(s[j + 1 : j  + lenWord + 1])
            i = j + lenWord + 1
        return res


# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))

File: C++/encode_and_decode_strings.cpp
class Codec {
public:

    // Encodes a list of strings to a single string.
    string encode(vector<string>& strs) {
        string res = "";

        for (string s : strs){
            res += to_string(s.size()) + '#' + s;
        }
        return res;
    }

    // Decodes a single string to a list of strings.
    vector<string> decode(string s) {
        int i = 0;
        vector<string> res;

        while (i < s.size()){
            int j = i;
            while (s[j] != '#'){
                j++;
            }

            int wordLen = stoi(s.substr(i, j - i));
            res.push_back(s.substr(j + 1, wordLen));
            i = j + 1 + wordLen;
        }
        return res;
    }
};

// Your Codec object will be instantiated and called as such:
// Codec codec;
// codec.decode(codec.encode(strs));
